runtime/js/01_async_context.js

```diff
@@ -0,0 +1,45 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+import { primordials } from "ext:core/mod.js";
+import { op_get_extras_binding_object } from "ext:core/ops";
+
+const {
+  SafeWeakMap,
+} = primordials;
+
+const {
+  getContinuationPreservedEmbedderData,
+  setContinuationPreservedEmbedderData,
+} = op_get_extras_binding_object();
+
+let counter = 0;
+
+export const getAsyncContext = getContinuationPreservedEmbedderData;
+export const setAsyncContext = setContinuationPreservedEmbedderData;
+
+export class AsyncVariable {
+  #id = counter++;
+  #data = new SafeWeakMap();
+
+  enter(value) {
+    const previousContextMapping = getAsyncContext();
+    const entry = { id: this.#id };
+    const asyncContextMapping = {
+      __proto__: null,
+      ...previousContextMapping,
```

Comment:
Q: Can `__proto__` be overriden by `previousContextMapping` in theory? I'm not really familiar with the async API like if it is a user supplied value or not.

Reply:
it cannot